src/amoc_config/amoc_config.hrl

```diff
@@ -0,0 +1,60 @@
+%%==============================================================================
+%% Copyright 2020 Erlang Solutions Ltd.
+%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
+%%==============================================================================
+
+
+%%------------------------------------------------------------------------------
+%% basic types
+%%------------------------------------------------------------------------------
+-type name() :: atom().
+-type value() :: any().
+
+%%------------------------------------------------------------------------------
+%% runtime supplied settings
+%%------------------------------------------------------------------------------
+-type parameter() :: {name(), value()}.
+-type settings() :: [parameter()].
+
+%%------------------------------------------------------------------------------
+%% error types
+%%------------------------------------------------------------------------------
+-type reason() :: any().
+-type error_type() :: atom().
+-type error() :: {error, error_type(), reason()}.
+
+%%------------------------------------------------------------------------------
+%% module attributes definition
+%%------------------------------------------------------------------------------
+-type verification_fun() :: fun((Value :: value()) -> boolean() |
+                                                      {true, NewValue :: value()} |
+                                                      {false, reason()}).
```

Comment:
this is not a new format, i just moved it into this header. boolean makes perfect sense as these functions are a kind of guards. also it allows using erlang guard functions as is. 
I'm not going to change it since it's already used in amoc-arsenal-xmpp.